develop/parachain-devs/runtime-development/FRAME/create-custom-pallet.md

```diff
@@ -0,0 +1,334 @@
+
+Custom pallets offer several advantages:
+
+- Flexibility - define runtime behavior that precisely matches your project requirements
+- Modularity - combine pre-built and custom pallets to achieve the desired blockchain functionality
+- Scalability - add or modify features as your project evolves
+
+As you follow this guide to create your custom pallet, you'll work with the following key sections:
+
+1. Imports and dependencies - bring in necessary FRAME libraries and external modules
+2. Runtime configuration trait - specify the types and constants required for your pallet to interact with the runtime
+3. Runtime events - define events that your pallet can emit to communicate state changes
+4. Runtime errors - define the error types that can be returned from the function calls dispatched to the runtime
+5. Runtime storage - declare on-chain storage items for your pallet's state
+6. Extrinsics (function calls) - create callable functions that allow users to interact with your pallet and execute transactions
+
+## Initial Setup
+
+This section will guide you through the initial steps of creating the foundation for your custom FRAME pallet. You'll create a new Rust library project and set up the necessary dependencies.
+
+1. Create a new Rust library project
+
+    Execute the following command using `cargo`:
+
+    ```bash
+    cargo new --lib custom-pallet \
+    && cd custom-pallet
+    ```
+
+    This command creates a new library project named `custom-pallet` and navigates into its directory.
+
+2.  Configure dependencies in `Cargo.toml` file
+
+    Add the core dependencies required for FRAME pallet development. Open the `Cargo.toml` file and update it as follows:
+
+    ```toml
+    [package]
+    name = "custom-pallet"
+    version = "0.1.0"
+    edition = "2021"
+    
+    [dependencies]
+    frame-support = { version = "37.0.0", default-features = false }
+    frame-system = { version = "37.0.0", default-features = false }
+    codec = { version = "3.6.12", default-features = false, package = "parity-scale-codec", features = ["derive"] }
+    scale-info = { version = "2.11.1", default-features = false, features = ["derive"] }
+    sp-runtime = { version = "39.0.0", default-features = false }
+
+    [features]
+    default = ["std"]
+    std = [
+        "frame-support/std",
+        "frame-system/std",
+        "codec/std",
+        "scale-info/std",
+        "sp-runtime/std",
+    ]
+    ```
+
+    !!!note
+        Carefully select the versions of the packages according to your project's specific requirements:
+
+        - When developing for a specific Polkadot SDK runtime, ensure that your pallet's dependency versions match those of the target runtime
+        - If you're creating this pallet within a Polkadot SDK workspace:
+
+            - Define the actual versions in the root `Cargo.toml` file
+            - Use workspace inheritance in your pallet's `Cargo.toml` to maintain consistency across your project
```

Comment:
I'd add a sentence that refers to the page "Add Existing Pallet to the Runtime" since there it's explained this workspace inheritance thing